### PR TITLE
Allow <Text /> nodes to be used and translated in template field values

### PR DESCRIPTION
**Pull request by billneff79**
resolves issue #29

Built size goes from 1365 bytes to 1402 bytes (+37 bytes)

### CODE REVIEW

File: src/lib/template.js [hunk 1]
```diff
@@ -10,20 +11,24 @@ let currentFields;
  *	@example
  *		template('foo{{bar}}', { bar:'baz' }) === 'foobaz'
  */
-export default function template(template, fields) {
-	currentFields = fields || EMPTY;
-	return template && template.replace(/\{\{([\w.-]+)\}\}/g, replacer);
+export default function template(template, fields, scope, dictionary) {
+	return template && template.replace(/\{\{([\w.-]+)\}\}/g, replacer.bind(null, fields || EMPTY, scope, dictionary));
```
Comment (billneff79) [inline src/lib/template.js]: while this forces a `bind` to run on every instance, that isn't expensive and is much safer than using a file-scoped variable that changes on every instantiation

File: test/lib/translate.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { h } from 'preact';
+import translate from '../../src/lib/translate';
+import { Text } from '../../src/components/text';
+
+describe('translate', () => {
```
Comment (billneff79) [inline test/lib/translate.js]: this isn't a complete testing of `translate`, mostly missing out on the `plural` test cases, but those are covered already in the testing for `<Text />`.

**Comment by billneff79** [comment]
@pl12133 can you take a look at this one?

File: README.md [hunk 3]
```diff
@@ -209,7 +209,7 @@ It "renders" out any `<Text />` values in its child's props.
 
 -   `props` **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** props
     -   `props.id` **[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** Key to look up in intl dictionary, within any parent scopes (`$scope1.$scope2.$id`)
-    -   `props.fields` **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** Values to inject into template `{{fields}}` (optional, default `{}`)
+    -   `props.fields` **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** Values to inject into template `{{fields}}`.  Values in the fields object will be interpreted as strings, except `<Text />` nodes which will be interpreted/translated (optional, default `{}`)
```
Comment (pl12133) [inline README.md]: The word "interpreted" twice here might be confusing. The second "interpreted/translated" could become "resolved".

Comment (pl12133) [reply README.md]: I like the sound of that.

Comment (billneff79) [reply README.md]: I updated the text with that value in the latest commit

**Pull request merged**